src/NLog.Extensions.Logging/Config/NLogLoggingConfiguration.cs

```diff
@@ -46,64 +46,80 @@ private void LoadConfigurationSection(IConfigurationSection nlogConfig, bool? au
 
         private class LoggingConfigurationElement : ILoggingConfigurationElement
         {
-            readonly IConfigurationSection _configurationSection;
-            readonly string _nameOverride;
+            private const string VariablesKey = "Variables";
+            private const string VariableKey = "Variable";
+            private const string TargetKey = "target";
+            private readonly IConfigurationSection _configurationSection;
+            private readonly string _nameOverride;
+            private readonly bool _topElement;
 
+            public string Name => _nameOverride ?? _configurationSection.Key;
+            public IEnumerable<KeyValuePair<string, string>> Values => GetValues();
+            public IEnumerable<ILoggingConfigurationElement> Children => GetChildren();
             public bool AutoReload { get; }
 
             public LoggingConfigurationElement(IConfigurationSection configurationSection, bool topElement, string nameOverride = null)
             {
                 _configurationSection = configurationSection;
                 _nameOverride = nameOverride;
-                if (topElement && bool.TryParse(configurationSection["autoreload"], out var autoreload))
+                _topElement = topElement;
+                if (topElement)
                 {
-                    AutoReload = autoreload;
+                    if (bool.TryParse(configurationSection["autoreload"], out var autoreload))
```

Comment:
PS. This will create a sonar alert ;)

Reply:
Never understood that warning about having as many conditions as possible on one line.